### PR TITLE
feature: webview custom dialog

### PR DESCRIPTION
**Pull request by minh-rakuten**
# Description
- Override js dialog event to not show URL.
- The action texts of the native dialog are shown as localized strings.
- Show miniapp display name in dialog title. This made the changes in source code:
Pass `MiniAppInfo` to webview. Remove `miniapp id`.
Save `MiniAppInfo` as json string in local.  Remove `version id` in storage.

## Links
[MINI-1228](https://jira.rakuten-it.com/jira/browse/MINI-1228)

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Comment by corycaywood** [comment]
It's looking good, but I think we should be able to inject the Mini App title into the webview.

I'm thinking that instead of fetching the latest `MiniAppInfo` from within `MiniAppDownloader`, you could instead move this to `RealMiniApp#executingCreate` (using `miniAppInfoFetcher.getInfo(miniAppId)`). And then we can inject the `MiniAppInfo` into the the call to `Displayer#createMiniAppDisplay`. What do you think?

**Comment by minh-rakuten** [comment]
> It's looking good, but I think we should be able to inject the Mini App title into the webview.
> 
> I'm thinking that instead of fetching the latest `MiniAppInfo` from within `MiniAppDownloader`, you could instead move this to `RealMiniApp#executingCreate` (using `miniAppInfoFetcher.getInfo(miniAppId)`). And then we can inject the `MiniAppInfo` into the the call to `Displayer#createMiniAppDisplay`. What do you think?

@corycaywood We can let the `MiniAppDownloader#getMiniApp` return pair of values `Pair<MiniAppPath: String, miniappInfo: MiniAppInfo>` and we can pass that `MiniAppInfo` to `Displayer#createMiniAppDisplay`.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/WebViewDialog.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+internal fun onShowDialog(
+    context: Context,
+    message: String?,
+    defaultValue: String? = "",
+    result: JsResult?,
+    dialogType: DialogType,
+    miniAppInfo: MiniAppInfo
+): Boolean {
+    val dialogBuilder = AlertDialog.Builder(context)
+    dialogBuilder.setTitle(miniAppInfo.displayName).setMessage(message)
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/WebViewDialog.kt:23]:
```suggestion
    val dialogBuilder = AlertDialog.Builder(context)
        .setTitle(miniAppInfo.displayName)
        .setMessage(message)
```

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt [hunk 2]
```diff
@@ -55,20 +60,16 @@ internal class MiniAppDownloader(
...
+        val versionId = miniAppInfo.version.versionId
         val baseSavePath = storage.getMiniAppVersionPath(appId, versionId)
         when {
             isManifestValid(manifest) -> {
                 for (file in manifest.files) {
                     val response = apiClient.downloadFile(file)
                     storage.saveFile(file, baseSavePath, response.byteStream())
                 }
-                miniAppStatus.setVersionDownloaded(appId, versionId, true)
-                miniAppStatus.saveDownloadedVersion(appId, versionId)
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt:71]: Just curious to know, could we call `storeDownloadedMiniApp(miniAppInfo)` here?

Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt:71]: There is the reason why I changed the call of this method in `getMiniApp` function.

We cached only `versionId` before and now replace it with `miniAppInfo`. Our SDK checks that there is cache but no `miniAppInfo` saved so we need to update the status of downloaded miniapp even it's already downloaded.

**Pull request merged**